### PR TITLE
Allow ignoring paths.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #707

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me. What do you think @saeideng ?

File: docs/en/index.rst [hunk 1]
```diff
@@ -50,6 +50,12 @@ Configuration
     // Before loading DebugKit
     Configure::write('DebugKit.forceEnable', true);
 
+* ``DebugKit.ignorePaths`` - Regex pattern (including delimiter) to ignore paths.
+   DebugKit won't save data for request URLs which match this regex. Defaults to ``null``::
```
Comment (othercorey) [inline docs/en/index.rst:54]:
```suggestion
   DebugKit won't save data for request URLs that match this regex. Defaults to ``null``::
```
https://www.grammarly.com/blog/which-vs-that/ (not to be pretentious, it's just a good explanation of why `which` feels wrong)

File: tests/TestCase/ToolbarServiceTest.php [hunk 2]
```diff
@@ -164,6 +164,40 @@ public function testSaveDataIgnoreDebugKitDashedUrl()
         $this->assertFalse($bar->saveData($request, $response));
     }
 
+    /**
+     * Test that saveData ignores path which match "ignorePaths" regex.
```
Comment (othercorey) [inline tests/TestCase/ToolbarServiceTest.php:168]:
```suggestion
     * Test that saveData ignores path that matches "ignorePaths" regex.
```
just being thorough

File: src/ToolbarService.php [hunk 3]
```diff
@@ -67,6 +67,7 @@ class ToolbarService
         ],
         'forceEnable' => false,
         'safeTld' => [],
+        'ignorePaths' => null,
```
Comment (othercorey) [inline src/ToolbarService.php:70]: Maybe this is cakephp convention, but `ignorePaths` doesn't feel like a pattern to me.

Up to you, it's not a major api.

File: docs/en/index.rst (hunk 1 above)
Comment (ADmad) [reply docs/en/index.rst]: I actually appreciate the English grammar lesson :slightly_smiling_face:.

File: src/ToolbarService.php (hunk 3 above)
Comment (ADmad) [reply src/ToolbarService.php:70]: We could add a `Regex` suffix to the option name but I don't remember similar being done elsewhere.

Comment (othercorey) [reply src/ToolbarService.php:70]: I was thinking `ignorePathsPattern` but whatever. If this is convention then it's convention.

**Comment by saeideng** [comment]
> Looks good to me. What do you think @saeideng ?

I tested it locally, it works well 
good job @ADmad :+1:

**Pull request merged**